pref-adjust.lua

```diff
@@ -269,6 +268,51 @@ function build_all_lists(printflag)
     end
 end -- end func build_all_lists
 -- ---------------------------------------------------------------------------
+function get_preferences(unit)
+    if unit == nil then
+        print("No unit selected!")
+        return
+    end
+
+    local preferences = unit.status.current_soul.preferences
+    if #preferences == 0 then
+        print("Unit " .. unit_name_to_console(unit) .. " has no preferences.")
+        return
+    end
+
+    print("Preferences for " .. unit_name_to_console(unit) .. ":")
+    for _, pref in ipairs(preferences) do
+        local pref_type = df.unit_preference.T_type[pref.type]
+        local description = ""
+
+        if pref_type == "LikeMaterial" then
+            description = "Likes material: " .. dfhack.matinfo.getToken(pref.mattype, pref.matindex)
+        elseif pref_type == "LikeFood" then
+            description = "Likes food: " .. dfhack.matinfo.getToken(pref.mattype, pref.matindex)
+        elseif pref_type == "LikeItem" then
+            description = "Likes item type: " .. tostring(pref.item_type)
+        elseif pref_type == "LikePlant" then
+            description = "Likes plant: " .. dfhack.matinfo.getToken(pref.mattype, pref.matindex)
+        elseif pref_type == "HateCreature" then
+            description = "Hates creature: " .. df.global.world.raws.creatures.all[pref.poetic_form_id].creature_id
```

Comment:
```suggestion
            description = "Hates creature: " .. df.global.world.raws.creatures.all[pref.creature_id].creature_id
```

Reply:
the existing code only writes to one field of the union, but you can read the appropriate fields here